tests/httpfake.h

```diff
@@ -110,11 +110,12 @@ class HttpFake : public HttpInterface {
   }
 
   HttpResponse put(const std::string &url, const Json::Value &data) override {
-    (void)data;
+    if (url.find("putmanifesterror") != std::string::npos) {
+      return HttpResponse(url, 504, CURLE_OK, "");
+    }
```

Comment:
Laurent recently did some work to try to remove most of the test-case-specific code out of httpfake.h itself and instead derive a new class that overloads the one thing necessary. Could you do that instead? There should be plenty of examples in uptane_test.cc.